### PR TITLE
Add Ubuntu Support

### PR DESCRIPTION
**Pull request by razorsedge**
Adding Ubuntu and Debian support to (most) all of the scripts.

### CODE REVIEW

File: api/install_dump_cm_config.sh [hunk 1]
```diff
@@ -14,16 +14,59 @@
...
 
-# https://discourse.criticalengineering.org/t/howto-password-generation-in-the-gnu-linux-cli/10
-PWCMD='< /dev/urandom tr -dc A-Za-z0-9 | head -c 20;echo'
-#if ! rpm -q apg; then echo "Installing apg. Please wait...";yum -y -d1 -e1 install apg; fi
-#if rpm -q apg; then export PWCMD='apg -a 1 -M NCL -m 20 -x 20 -n 1'; fi
-if ! rpm -q apg >/dev/null; then
-  echo "Installing apg. Please wait..."
-  yum -y -d1 -e1 install apg
+# Function to discover basic OS details.
+discover_os () {
```
Comment (Shekharv) [inline api/install_dump_cm_config.sh]: This is one of the things with bash - discover_os is in two different places.. pretty much the same code?

File: change_swappiness.sh [hunk 2]
```diff
@@ -16,21 +16,54 @@
 
 VAL=1
 
-if rpm -q redhat-lsb-core; then
-  OSREL=`lsb_release -rs | awk -F. '{print $1}'`
-else
-  OSREL=`rpm -qf /etc/redhat-release --qf="%{VERSION}\n"`
+# Function to discover basic OS details.
+discover_os () {
```
Comment (Shekharv) [inline change_swappiness.sh]: discover_os again.. I am no bash ninja - but doesn't this make life difficult? changing this logic in so many places if required.

Comment (razorsedge) [inline change_swappiness.sh]: Each shell script is standalone.  I might refactor to source a central script that holds this function, but then users have to make sure they copy the central script as well.

File: api/install_dump_cm_config.sh (hunk 1 above)
Comment (razorsedge) [inline api/install_dump_cm_config.sh]: Yes.  Its copied in most of the scripts.

Comment (Shekharv) [inline api/install_dump_cm_config.sh]: OK sounds good. See if we can create a JIRA and have someone else volunteer (or assigned ) to do that? Would force more familiarity? you can review the PR and ensure it works

Comment (razorsedge) [inline api/install_dump_cm_config.sh]: HAM-621 created.

**Pull request merged**